src/components/Article.jsx

```diff
@@ -6,6 +6,37 @@ import Comments from "./Comments";
 function Article() {
     const { article_id } = useParams()
     const [article, setArticle] = useState({})
+    const [votePending, setVotePending] = useState(false)
+
+    function handleUpVote(){
+
+        setVotePending(true)
+
+        const newVotes = article.votes ++
+        setArticle({...article, newVotes})
+        axios.patch(`https://backend-project-c921.onrender.com/api/articles/${article_id}`, {inc_votes: 1})
+        .then(()=>{
+            setVotePending(false)
+        })
+        .catch(() =>{
+            setVotePending(false)
+            alert(`Vote Failed on article: \n\n ${article.title}. \n\n Please retry`)
+        })
+    }
+
+    function handleDownVote(){
```

Comment:
The only difference between these functions are the increment or decrement you might be able to make a reusable function here